modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutor.java

```diff
@@ -70,6 +71,9 @@
     /** Index of alive node URI. */
     private Map<List<String>, Integer> startIdxs = U.newHashMap(2);
 
+    /** Cache for parsed URLs. */
+    private Map<String, HttpUrl> urlCache = new ConcurrentHashMap<>();
```

Comment:
I think lazy cache of parsed url is not good idea. Why we don't map nodeURIs to Collection<HttpUrl> before any sendRequest?

Reply:
Will rollback, as it is not that easy to refactor.